crypto/crypto.go

```diff
@@ -70,7 +71,19 @@ type KeccakState interface {
 
 // NewKeccakState creates a new KeccakState
 func NewKeccakState() KeccakState {
-	return sha3.NewLegacyKeccak256().(KeccakState)
+	state := hasherPool.Get().(KeccakState)
```

Comment:
It's a bit weird to apply the sync Pool here. It implies that `ReturnKeccakState` must be paired with this function.

I would prefer to only use syncPool in `Keccak256` and `Keccak256Hash`.